prov/efa/test/efa_unit_test_info.c

```diff
@@ -0,0 +1,42 @@
+#include "efa_unit_tests.h"
+
+/**
+ * @brief test that when a wrong fi_info was used to open resource, the error is handled
+ * gracefully
+ */
+void test_info_open_ep_with_wrong_info()
+{
+	struct fi_info *hints, *info;
+	struct fid_fabric *fabric = NULL;
+	struct fid_domain *domain = NULL;
+	struct fid_ep *ep = NULL;
+	int err;
+
+	hints = efa_unit_test_alloc_hints(FI_EP_DGRAM);
+
+	err = fi_getinfo(FI_VERSION(1, 14), NULL, NULL, 0ULL, hints, &info);
+	assert_int_equal(err, 0);
+
+	/* dgram endpoint require FI_MSG_PREFIX */
+	assert_int_equal(info->mode, FI_MSG_PREFIX);
+
+	/* make the info wrong by setting the mode to 0 */
+	info->mode = 0;
+
+	err = fi_fabric(info->fabric_attr, &fabric, NULL);
+	assert_int_equal(err, 0);
+
+	err = fi_domain(fabric, info, &domain, NULL);
+	assert_int_equal(err, 0);
+
+	/* because of the error in the info object, fi_endpoint() should fail with -FI_ENODATA */
+	err = fi_endpoint(domain, info, &ep, NULL);
+	assert_int_equal(err, -FI_ENODATA);
```

Comment:
Should we make the return code `FI_EINVAL` instead?

`FI_ENODATA` is odd - why does the program complain about no data trying to open an endpoint.